### ISSUE TITLE
`.reversePens()` treats glyph ordering differently, when using multiline in Glyphwise

### ISSUE THREAD

**Issue by Mark2Mark** [coldtype/coldtype#108]
As soon, as the Glyphwise utilizes a `\n` for multiline, the `reversePens()` reorders the entire words instead of the single glyphs, as it does when there is only a  one-line-string.

This might be desired or a bug. If it’s desired, how can one achieve the glyph reordering in a multiline Glyphwise string?

compare:

```py
# A
@aframe((1080, 1080), bg=0)
def reverse_as_expected(f):
    return P(
            Glyphwise("COLD", lambda g:
            Style(Font.ColdObvi(), 300, wght=1, tu=-100))
            .align(f.a.r)
            .understroke()
            .reversePens()
            )

# B
@aframe((1080, 1080), bg=0)
def reverse_not_as_expected(f):
    return P(
            Glyphwise("COLD\nTYPE", lambda g:
            Style(Font.ColdObvi(), 300, wght=1, tu=-100))
            .align(f.a.r)
            .understroke()
            .reversePens()
            )
```

A
<img width="538" alt="Screenshot 2022-06-12 at 12 18 50" src="https://user-images.githubusercontent.com/5824595/173228656-e3ee2c0e-1b0a-4c30-9111-6cf47bcd8b9b.png">

B
<img width="538" alt="Screenshot 2022-06-12 at 12 19 02" src="https://user-images.githubusercontent.com/5824595/173228653-3d13277c-2a7d-4bd8-8d9d-5e07f7b0d36e.png">

PS: Is there a difference between `reversePens()` and `reverse()`?

**Comment by stenson** [comment]
Great question! `reversePens()` used to be distinct from `reverse()`, though now they are the same function, I think for some reason I just got used to writing `reversePens()` so its use has kind of persisted in the example for too long. That said, `reverse()` also takes a optional argument, `recursive=`, which could be one potential solution to the multiline issue you're talking about, like this:

```python
@aframe((1080, 1080), bg=0)
def reverse_singleline_recursive(f):
    return P(
            Glyphwise("COLD", lambda g:
                Style(Font.ColdObvi(), 300, wght=1, tu=-100))
            .align(f.a.r)
            .reverse(recursive=True)
            .fssw(1, 0, 20, 1) # alternate to understroke, (fssw with four arguments translates to fill, stroke, strokewidth, strokefirst — the final strokefirst argument means the shape is "stroked" before it’s filled, meaning the fill partially overlays the stroke, unless like most default graphic engine behaviors, this is particularly useful instead of `understroke` because you're sending half as many distinct shapes to the renderer, so it should be quite a bit faster)
            .print())
```

With that code, a single-line and multi-line Glyphwise structure would be reversed in the same way (lines reversed and glyphs reversed in each line).

The downside to this approach is that, because `recursive` means `reverse` is called recursively on the whole data structure, this call also reverses the winding direction of the individual glyphs themselves, since a `reverse()` call on a plain glyph/path object reverses winding direction (via a `ReverseContourPen`). Not a big deal in most cases I don't think, since the winding direction change wouldn't matter unless you were flattening a Glyphwise into a single pen and combining it with some other pen via an intersection or something like that, which would mean you'd lose the overlapping glyphs anyway.

To more "manually" reverse elements in a multi-line Glyphwise structure, you need to add a `.map(lambda p: p.reverse())` call after the `reverse()` call, i.e. call reverse on the lines as a unit, then on each line (via map), like this:

```python
@aframe((1080, 1080), bg=0)
def reverse_multiline_literal(f):
    return P(
        Glyphwise("COLD\nTYPE", lambda g:
            Style(Font.ColdObvi(), 300, wght=1, tu=-100))
        .align(f.a.r)
        #.reverse(recursive=True)
        .reverse()
        .map(lambda p: p.reverse())
        .fssw(1, 0, 20, 1))
```

All that said, unlike `Glyphwise`, `StSt` accepts a `multiline=` kwarg that lets you "force" the result to be structured as if there was a "\n" present, no matter what, i.e. you get a nested multi-line style result, even if you only pass one line of text in. I only bring this up to say, `Glyphwise` should definitely support the same kwarg, so I'll close this issue when that change gets released.

**Comment by Mark2Mark** [comment]
Awesome! Thank you so much. I hope it was okay that I opened an issue, always only afterwards I think it might have had a better place in the discussions instead.

Will use the .map-lambda approach, as this makes most sense to me in terms of sticking to the Coldtype syntax style :)

Thanks a million!

**Comment by stenson** [comment]
`multiline` kwarg on `Glyphwise` now available in 0.9.7
